### PR TITLE
Update README.md

### LINKED ISSUE #2668: Rework the docs and README after renaming datasets-server -> dataset-viewer
Now that we call the project "Dataset viewer", we should make it clear in the docs and READMEs that:

- that the project powers the Hub's dataset viewer
  - people can file issues if something is broken in the Hub dataset viewer
  - contributions here will help improving the dataset viewer
- that it does not contain source code for the viewer frontend

### PR DESCRIPTION
**Pull request by severo**
fixes #2668

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,22 +1,27 @@
...
+Documentation:
+- dataset viewer: https://huggingface.co/docs/hub/datasets-viewer
+- configuration of the datasets: https://huggingface.co/docs/hub/datasets-data-files-configuration
+- backend's API: https://huggingface.co/docs/dataset-viewer
 
-We plan to [add more features](https://github.com/huggingface/dataset-viewer/issues?q=is%3Aissue+is%3Aopen+label%3A%22feature+request%22) to the server. Please comment there and upvote your favorite requests.
+## You saw a bug 🪲 or want a new feature 🎁
 
-If you think about a new feature, please [open a new issue](https://github.com/huggingface/dataset-viewer/issues/new).
+If the dataset viewer is showing an error on your dataset page, please [open a discussion](https://huggingface.co/docs/hub/repositories-pull-requests-discussions) there, it's the most efficient way to fix it.
```
Comment (polinaeterna) [inline README.md:16]: maybe also specify whom to tag to get help?

**Review by polinaeterna — APPROVED** [review]
awesome, thank you!

**Review by julien-c — COMMENTED** [review]
nice, it's clearer

File: README.md (hunk 1 above)
Comment (severo) [reply README.md]: good idea. Done

**Pull request merged**